test/testPartyAttachments.js

```diff
@@ -0,0 +1,1025 @@
+const assert = require("assert");
+
+const FormData = require('form-data');
+const uuid = require("uuid");
+const { debug } = require("../js/common.js");
+const fs = require("fs");
+
+/**
+ * @typedef { {
+ *  remotefileName: string,
+ *  urlToCopy?: string,
+ *  localFilename: string,
+ *  attachmentKey: string,
+ *  resourceHref: string
+ * } } TFileToUpload
+ * @typedef { {
+ *  remotefileName: string,
+ *  urlToCopy: string,
+ *  attachmentKey: string,
+ *  resourceHref: string
+ * } } TFileToCopy
+ * 
+ * @typedef { import("./httpClient.js").THttpClient } THttpClient
+ * @typedef { import("./httpClient.js").THttpResponse } THttpResponse
+ * @typedef { import("../js/sri4node-attachments.js").TReadableStream } TReadableStream
+ */
+
+/**
+ * This function does a GET requests as a streaming request and when succesfull, returns
+ * a stream of the body.
+ * @param {THttpClient} httpClient
+ * @param {string} url 
+ * @returns {Promise<TReadableStream>}
+ */
+const doGetStream = async (httpClient, url) => {
+    const { status, body } = await httpClient.get({ path: url, streaming: true });
+    assert.equal(status, 200, 'getStream received unexpected status code');
+    return body;
+};
+
+/**
+ * This function requests deletion of an attachments and then verifies if the attachments
+ * is really gone.
+ * @param {THttpClient} httpClient 
+ * @param {string} attachmentUrl 
+ * @param {string} attachmentDownloadUrl 
+ */
+const deleteAttachmentAndVerify = async (
+  httpClient,
+  attachmentUrl,
+  attachmentDownloadUrl
+) => {
+  // Delete the attachment
+  const responseDelete = await httpClient.delete({ path: attachmentUrl });
+  assert.equal(responseDelete.status, 204);
+
+  // verify if attachment is gone
+  const responseGetAtt2 = await httpClient.get({ path: attachmentUrl });
+  assert.equal(responseGetAtt2.status, 404);
+  const responseGetAtt3 = await httpClient.get({ path: attachmentDownloadUrl });
+  assert.equal(responseGetAtt3.status, 404);
+};
+
+
+/**
+ * Creates the JSON body describing the attachment files, which can be passed at multipart
+ * upload or at /attachments/copy ?
+ * @param {Array<TFileToUpload | TFileToCopy>} fileDetails 
+ * @returns 
+ */
+const createUploadBody = (fileDetails) => {
+  return fileDetails.map(
+    ({ remotefileName, attachmentKey, resourceHref, urlToCopy }) => ({
+      file: remotefileName,
+      fileHref: urlToCopy, // can be undefined, but if it's there, this is the url that should be copied
```

Comment:
This probably doesn't cause any issues but it is a bit weird to add an undefined fileHref so I would add it conditionally, eg like this: https://stackoverflow.com/a/40560953/9266796.

Reply:
To me this is more clear then conditionally adding fileHref.